src/main/java/tweeting/resources/TwitterResource.java

```diff
@@ -0,0 +1,127 @@
+            return service.postTweet(message)
+                    .map(timeline -> Response.status(Response.Status.CREATED)
+                            .entity(timeline)
+                            .build())
+                    .orElseThrow(() -> new NullPointerException("Twitter failed to respond with home timeline."));
+        } catch (TwitterServiceCallException e) {
+            logger.debug("Sending 400 Bad Request error", e);
+            return Response.status(Response.Status.BAD_REQUEST)
+                    .entity(e.getMessage())
+                    .build();
+        } catch (TwitterServiceResponseException e) {
+            logger.error("Sending 500 Internal Server error", e);
+            return (Response.status(Response.Status.INTERNAL_SERVER_ERROR)
+                    .entity(e.getMessage()))
+                    .build();
+        } catch (Exception e) {
+            logger.error(e.getMessage(), e);
+            return (Response.status(Response.Status.INTERNAL_SERVER_ERROR)
+                    .entity(TwitterService.SERVICE_UNAVAILABLE_MESSAGE))
+                    .build();
+        }
+    }
+
+    /*
+     * How to use:
+     * curl -i -X GET http://HOST:PORT/api/1.0/twitter/timeline
+     *
+     * Replace HOST and PORT with configured values
+     */
+    @Path("twitter/timeline")
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getHomeTimeline() {
+        try {
+            return service.getHomeTimeline()
+                    .map(timeline -> Response.ok(timeline)
+                            .build())
+                    .orElseThrow(() -> new NullPointerException("Twitter failed to respond with home timeline."));
+        } catch (TwitterServiceResponseException e) {
+            logger.error("Sending 500 Internal Server error", e);
+            return (Response.status(Response.Status.INTERNAL_SERVER_ERROR)
+                    .entity(e.getMessage()))
+                    .build();
+        } catch (Exception e) {
+            logger.error(e.getMessage(), e);
+            return (Response.status(Response.Status.INTERNAL_SERVER_ERROR)
+                    .entity(TwitterService.SERVICE_UNAVAILABLE_MESSAGE))
+                    .build();
+        }
+    }
+
+    /*
+     * How to use:
+     * curl -i -X GET http://HOST:PORT/api/1.0/tweet/filter?keyword=KEYWORD
+     *
+     * Replace 'KEYWORD' with desired keyword to filter by. Replace HOST and PORT with configured values
+     */
+    @Path("tweet/filter")
+    @GET
+    @Consumes(MediaType.APPLICATION_FORM_URLENCODED)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getFilteredHomeTimeline(@QueryParam("keyword") String keyword) {
+        try {
+            return service.getFilteredTimeline(keyword)
+                    .map(timeline -> Response.ok(timeline)
+                            .build())
+                    .orElseThrow(() -> new NullPointerException("Twitter failed to respond with home timeline."));
```

Comment:
this feels odd to just throw it below, if this is the case, why not just have the other class throw the exception?